lib/learn_open/environments/base_environment.rb

```diff
@@ -69,11 +69,29 @@ def notify_of_completion
       def warn_if_necessary(lesson)
         return unless lesson.later_lesson
 
+        warn_response = nil
+        prompt_count = 0
+        valid_responses = ['yes', 'y', 'no', 'n', "\n"]
+        visible = -> (token){ token.match(/\w+/) }
+        valid_resp_insert = [
+          *valid_responses.select{ |c| visible.call(c) },
+          "<ENTER>"
+        ].join(', ')
+
         io.puts 'WARNING: You are attempting to open a lesson that is beyond your current lesson.'
         io.print 'Are you sure you want to continue? [Y/n]: '
 
-        warn_response = io.gets.chomp.downcase
-        exit if !['yes', 'y'].include?(warn_response)
+        until valid_responses.include?(warn_response) or
+              prompt_count > 2
+          warn_response = io.gets.chomp.downcase
```

Comment:
Using chomp will kill the newline character if someone presses enter and replace with an empty string, and therefore causing another loop iteration. Maybe change this so that if there is an empty string, default to no.

Reply:
good catch!